rns_shell/compositor/layers/Layer.h

```diff
@@ -43,6 +46,56 @@ enum LayerInvalidateMask {
     LayerInvalidateAll = LayerPaintInvalidate | LayerLayoutInvalidate
 };
 
+struct ComponentShadow {
```

Comment:
Can add Api to get component to which shadow to be drawn is transparent or NOt as part of this structure:
bool isComponentTransparent() {
   return (opacity < 0xFF);
}

Reply:
As Discussed, No Need for Seperate API to get component transparency. But i see there many place !(opacity < 0xFF) is been used. It doesn't explain directly the purpose.
To make it meaning and handy. suggest to maintain a flag in Layer class to get the component transparency. and use it. That sounds more sensible for below reason 

- repeating this handling in many place "(opacity < 0xFF)"
- Passing bool is better than  float value

YOu can address this comment , in the end. Once you get a feedback cumulative feedback.
Summarizing Open Feedback Here from @pprabhu-ng :-

1. componentShadow, can be defined as Nested class for LAyer
2. structure Name componentShadoe to be changed, as it is part of Layer.